actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -580,7 +580,7 @@ def time_zone_options_for_select(selected = nil, priority_zones = nil, model = :
 
         if priority_zones
           if priority_zones.is_a?(Regexp)
-            priority_zones = zones.select { |z| z =~ priority_zones }
+            priority_zones = zones.select { |z| z.to_s =~ priority_zones }
```

Comment:
It is expected that instance of `ActiveSupport::TimeZone` be passed to `zones`. 
Since `ActiveSupport::TimeZone` has a `=~` method, so this warning should not normally show.
https://github.com/rails/rails/blob/0fa5b5510c6faefc33ab3715e08b2604195a1063/activesupport/lib/active_support/values/time_zone.rb#L331-L335 

The reason why this happens in the test is that the `=~` is not defined in the Fake class used in the test, so let's add the `=~` to the Fake class to deal with it.

Reply:
Thanks for the assistance! That did the trick.

Reply:
@y-yagi thanks for guidance! Wanted to contribute on this one, but was having a hard time figuring out the expected behavior for that particular test.